File: Serverside/Monolith Architecture/business_manager/src/main/java/com/IntegrityTechnologies/business_manager/modules/platform/tenant/service/TenantService.java
package com.IntegrityTechnologies.business_manager.modules.platform.tenant.service;

import com.IntegrityTechnologies.business_manager.modules.platform.tenant.dto.TenantCreateRequest;
import com.IntegrityTechnologies.business_manager.modules.platform.tenant.dto.TenantResponse;
import org.springframework.data.domain.Page;

import java.util.UUID;

public interface TenantService {

    TenantResponse createTenant(TenantCreateRequest request);

    TenantResponse getTenant(UUID id);

    Page<TenantResponse> getTenants(int page, int size);

    void deactivateTenant(UUID id);

}

File: Clientside/src/app/modules/platform/content/tenants/services/tenant.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { TenantResponse, TenantCreateRequest, TenantPage } from '../models/tenant.model';
import { environment } from '../../../../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class TenantService {

  private http = inject(HttpClient);
  private base = `${environment.apiUrl}/platform/tenants`;

  getTenants(
    page = 0,
    size = 20,
    search = ''
  ): Observable<TenantPage> {

    let url =
      `${this.base}?page=${page}&size=${size}`;

    if (search?.trim()) {
      url += `&search=${encodeURIComponent(
        search.trim()
      )}`;
    }

    return this.http.get<TenantPage>(url);
  }

  getTenant(id: string): Observable<TenantResponse> {
    return this.http.get<TenantResponse>(`${this.base}/${id}`);
  }

  createTenant(payload: TenantCreateRequest): Observable<TenantResponse> {
    return this.http.post<TenantResponse>(this.base, payload);
  }

  suspendTenant(id: string) {
    return this.http.post(`${environment.apiUrl}/platform/admin/tenants/${id}/suspend`, {});
  }

  activateTenant(id: string) {
    return this.http.post(`${environment.apiUrl}/platform/admin/tenants/${id}/activate`, {});
  }

  expireTenant(id: string) {
    return this.http.post(`${environment.apiUrl}/platform/admin/tenants/${id}/expire`, {});
  }

}
